src/db_api_util.ts

```diff
@@ -1031,6 +1052,67 @@ export async function getThreadNeedsAttentionFor(
   }
 }
 
+// Return a list of ThreadInfo's for all threads needing followup
+export async function getThreadsNeedingFollowUp(
+  slackUserId: string,
+  days: number
+): Promise<ThreadInfo[]> {
+  const client = await pool.connect();
+  try {
+    const result = await client.query(
+      `WITH all_claims AS (
```

Comment:
You can shrink the number of rows you use in the `WITH` clauses significantly if you use `SELECT DISTINCT ON`